### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by tarasavage**

### CODE REVIEW

File: src/arrayMethodSort.js [hunk 1]
```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
```
Comment (polosanya) [inline src/arrayMethodSort.js:7]:
```suggestion
  [].__proto__.sort2 = function(compareFunction = compareAsString) {
```
You can create it and set as default if nothing was passed

File: src/arrayMethodSort.js [hunk 2]
```diff
@@ -5,7 +5,35 @@
...
+        const stringX = String(x);
+        const stringY = String(y);
+
+        if (stringX > stringY) {
+          return 1;
+        } else if (stringX === stringY) {
+          return 0;
+        } else if (stringX < stringY) {
+          return -1;
+        }
```
Comment (polosanya) [inline src/arrayMethodSort.js:22]: You can move this logic to the separate function
```suggestion
      comparator = (x, y) => {
        const stringX = String(x);
        const stringY = String(y);

        if (stringX > stringY) {
          return 1;
        }
        
        if (stringX < stringY) {
          return -1;
        }
        
        return 0
```
Also, there is no need for `else if` as the code will not go further after `return` anyway

File: src/arrayMethodSort.js [hunk 3]
```diff
@@ -5,7 +5,35 @@
...
+      };
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
```
Comment (polosanya) [inline src/arrayMethodSort.js:27]: Do you really need two nested loops? I think only one is enough

**Review by polosanya — CHANGES_REQUESTED** [review]
Almost done! Let's improve your code a bit)

**Review by tetiana-ivanchenko — APPROVED** [review]

**Pull request closed without merging**